scrapy/core/downloader/webclient.py

```diff
@@ -7,17 +7,19 @@
 
 from scrapy.http import Headers
 from scrapy.utils.httpobj import urlparse_cached
+from scrapy.utils.python import to_bytes
 from scrapy.responsetypes import responsetypes
 
 
 def _parsed_url_args(parsed):
     path = urlunparse(('', '', parsed.path or '/', parsed.params, parsed.query, ''))
-    host = parsed.hostname
+    path = to_bytes(path)
+    host = to_bytes(parsed.hostname)
     port = parsed.port
-    scheme = parsed.scheme
-    netloc = parsed.netloc
+    scheme = to_bytes(parsed.scheme, encoding='ascii')
+    netloc = to_bytes(parsed.netloc)
     if port is None:
-        port = 443 if scheme == 'https' else 80
+        port = 443 if scheme == b'https' else 80
     return scheme, netloc, host, port, path
```

Comment:
Does this function assume parsed URL is ascii-only (because it went through `w3lib.url.safe_download_url`), so it is safe to use to_bytes with default utf8 encoding? Maybe it worths adding a comment about this assumption? URL encoding is tricky, and what all browsers do is just plain crazy: 

* URL path is encoded to utf-8, then percent-escaped;
* URL query arguments are encoded to the source page encoding, then percent-escaped;

It means joining path and query and then encoding them to utf8 is not compatible with other browsers unless the input was ascii-only.